Tasks/NuGetPublisher/nugetpublisher.ts

```diff
@@ -31,9 +31,11 @@ async function main(): Promise<void> {
 
         // read inputs
         let searchPattern = tl.getPathInput("searchPattern", true, false);
+        let failOnEmptyNupkgMatch = tl.getBoolInput("failOnEmptyNupkgMatch");
         let filesList = nutil.resolveFilterSpec(
             searchPattern,
-            tl.getVariable("System.DefaultWorkingDirectory") || process.cwd());
+            tl.getVariable("System.DefaultWorkingDirectory") || process.cwd(),
+            failOnEmptyNupkgMatch);
```

Comment:
The 3rd param to `resolveFilterSpec` is `allowEmptyMatch`. https://github.com/Microsoft/vsts-tasks/blob/master/definitions/nuget-task-common.d.ts#L12

Based on that, my understanding is that if `failOnEmptyNupkgMatch` is true, we should NOT allow empty matches, and vice versa, so we should pass in `!failOnEmptyNupkgMatch` (flip the bit to get the opposite value).